src/test/java/io/zentity/resolution/JobTest.java

```diff
@@ -109,6 +109,29 @@ public void testPopulateMatcherClauseIgnoreUnusedParams() throws Exception {
         Assert.assertEquals(matcherClause, expected);
     }
 
+    /**
+     * Sometimes the value extracted from the document contains quotations or backslashes that need to be escaped
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testPopulateMatcherClauseQuoteJsonString() throws Exception {
+        String matcherJson = "{\n" +
+                "  \"clause\": {\n" +
+                "    \"match\": {\n" +
+                "      \"{{ field }}\": \"{{ value }}\"\n" +
+                "    }" +
+                "  }\n" +
+                "}";
+        String nameAlias = "The \"One\"";
+        Matcher matcher = new Matcher("matcher_alias", matcherJson);
+        TreeMap<String, String> params = new TreeMap<>();
+        params.put("foo", "bar");
+        String matcherClause = Query.populateMatcherClause(matcher, "field_alias", nameAlias, params);
+        String expected = "{\"match\":{\"field_alias\":\"The \\\"One\\\"\"}}";
+        Assert.assertEquals(expected, matcherClause);
```

Comment:
I think ideally we should also verify that the result from `populateMatcherClause` is a valid json for all of these tests, WDYT?

Reply:
LGTM - Agreed, I'll write a test to check for valid JSON after merging this PR into the release branch